### DISCUSSION TITLE
Stop shipping minified code

### DISCUSSION THREAD

**Discussion by hipstersmoothie** [radix-ui/primitives#1092] [Ideas]
Hey! Love the components, they have saved me lots of time. 

The one thing that has made using them hard is that they come minified and obfuscated. While you do ship source maps it's still pretty hard to debug and I haven't been able to get them to work well. 

Would the team consider shipping unminified code so it's easier to debug?

**Comment by benoitgrelard** [comment]
@hipstersmoothie

This is something we've been struggling with if I'm honest, build pipeline etc isn't our main strength and we've fought a lot with parcel over the years and never found a good way to offer both dev/prod builds.

If you're familiar with this, we'd love some help!

**Reply by hipstersmoothie** [reply]
With designs systems I've maintained we always chose just to not minify our code. This make the process much simpler. We left it up to the consumer of our code to be minifying the code if they choose to. This way their dev builds are easy to debug and their prod builds are minified in a way they choose. 

So my plan for a PR would to be to just turn off the minification

**Reply by benoitgrelard** [reply]
I guess I just don't know enough as to why lots of libs provide both, react for example provide both.

**Reply by hipstersmoothie** [reply]
If you wanted to also ship minified code for a CDN that might make sense, but since the primitives are mainly react components you need a build step to use them anyway that might not be that useful.

**Reply by hipstersmoothie** [reply]
In react's case I think this is because they include a lot of stuff in the dev build they don't the prod build. If Radix really wanted to ship both dev and prod versions we could do something like the following 

https://unpkg.com/browse/react@16.7.0/index.js

**Reply by hipstersmoothie** [reply]
But since radix doesn't do anything special in dev vs prod I think it would be fine to just ship unminified code, letting the bundler do it's thing

**Reply by benoitgrelard** [reply]
> In react's case I think this is because they include a lot of stuff in the dev build they don't the prod build. If Radix really wanted to ship both dev and prod versions we could do something like the following
> 
> https://unpkg.com/browse/react@16.7.0/index.js

Yes, this is what we've been wanting/trying to do but we haven't been able to do so far.

> But since radix doesn't do anything special in dev vs prod I think it would be fine to just ship unminified code, letting the bundler do it's thing

We do though (prop types, warnings, DX stuff in general), and I think we'll do more and more of this.

**Reply by hipstersmoothie** [reply]
I started going down a react style route but I'm unsure how we could still deliver ESM code and still do that conditional exporting. I don't think you can do conditional exports and still expect tree shaking to work.

**Reply by devisales** [reply]
> @hipstersmoothie
> 
> This is something we've been struggling with if I'm honest, build pipeline etc isn't our main strength and we've fought a lot with parcel over the years and never found a good way to offer both dev/prod builds.
> 
> If you're familiar with this, we'd love some help!

What kind of problems have you guys been encountering with Parcel. We're thinking about using it to create our React customer facing component library. However we're thinking about rather to use a bundler or just TSC. Could you briefly share some issues you had with Parcel, if possible? Thanks!
